### PR TITLE
Fix wrong tooltip on charts with unaggregated data

### LINKED ISSUE #11907: Tooltip on unaggregated data does not show summed value like the visualization
**Describe the bug**
When showing a line/bar chart with unaggregated data, it sums up the values correctly in the visualization, but hovering the tick will only show a single data point in the tooltip.
Hovering the warning triangle in the right-top corner of the visualization will show the warning: `"d" is an unaggregated field: if it has more than one value at a point on the x-axis, the values will be summed.`

**To Reproduce**
Run this SQL (H2) and set the visualization to line/bar and hover the first tick, which has a summed value of `10` (shown on the visualization correctly), but will show `5` in the tooltip.
```
SELECT parsedatetime('2020-01-01', 'yyyy-MM-dd') AS "d", 5 AS "c" UNION ALL
SELECT parsedatetime('2020-01-01', 'yyyy-MM-dd') AS "d", 2 AS "c" UNION ALL
SELECT parsedatetime('2020-01-01', 'yyyy-MM-dd') AS "d", 3 AS "c" UNION ALL
SELECT parsedatetime('2020-01-02', 'yyyy-MM-dd') AS "d", 1 AS "c" UNION ALL
SELECT parsedatetime('2020-01-02', 'yyyy-MM-dd') AS "d", 4 AS "c"
```

**Expected behavior**
Tooltip should show same value as the visualization.

**Screenshots**
Screenshot_00000

**Information about your Metabase Installation:**
Metabase x.34.2 and `master` with multiple backends and browsers.

:arrow_down: Please click the :+1: reaction instead of leaving a `+1` or `update?` comment

### PR DESCRIPTION
**Pull request by alxnddr**
### Description

Fixes showing incorrect values in the tooltip when data is unaggregated.

The [previous attempt](https://github.com/metabase/metabase/pull/15472) to fix caused a regression on dashboard cards when multiple series are added.

Please approve the same fix for the master branch as well: https://github.com/metabase/metabase/pull/15783

#### Question 1 
Screen Shot 2021-04-23 at 02 12 45

#### Question 2
Screen Shot 2021-04-23 at 02 12 30

#### Dashboard
Correct tooltip for Question 1:
Screen Shot 2021-04-23 at 02 13 28

Correct tooltip for Question 2:
Screen Shot 2021-04-23 at 02 13 19

Fixes https://github.com/metabase/metabase/issues/11907

### CODE REVIEW

**Comment by nemanjaglumac** [comment]
So, the test for tooltips is already passing on `release-x.39.x` and the issue seems to be fixed. The regression was introduced in 38.4 and it somehow creeped in to the `master`, if I understood correctly?

That said, I am not sure which branch should this go to?
If it brings improvements for 39.1 - awesome. Include it here, but please open a follow up PR that is going to be a cherry-pick of this for the `master` to unblock it.

**Comment by alxnddr** [comment]
@nemanjaglumac yes, sorry about that. So the regression got in master from 0.38.4.

I created a separate PR for the master branch https://github.com/metabase/metabase/pull/15783

**Comment by nemanjaglumac** [comment]
No, no, no. Nothing to be sorry about. Thank YOU for dealing with this so quickly!

Great work.
I'd just wait for FE eng aproval. In the meantime, I'm building this locally and will give it a spin.

**Comment by nemanjaglumac** [comment]
Tested locally and everything seems to be working fine. since this fix directly fixes #11907, we should unskip the repro for it here.

However, let's please first wait for https://github.com/metabase/metabase/pull/15522 to get merged, then rebase this PR and unskip the 11907 repro. It's a bit messy but we need that throttle fix for Cypress tests. 11907 is failing for me locally because of it. Working like charm when I include the fix.

**Review by paulrosenzweig — APPROVED** [review]
Looks good to me!

Honestly, our tooltip code is so confusing, I don't trust visual inspection nearly as much as tests+QA. There's a lot of messiness in how we represent multiple dashboard series and breakouts series. Layering client aggregation logic on top is a lot! There are definitely much better ways to sequence the transforms than what we have now. As you get more comfortable, refactor with abandon!

One case to test if you haven't: what happens if we have a question with breakout series as one of the series on a multi-question dashboard card? Do client aggregations work? Probably not a super common case and not worth holding up this fix for, but it seems to live at the intersection of these features.

**Comment by alxnddr** [comment]
@paulrosenzweig I agree that it is a mess to put aggregation logic here. Another solution that does not require aggregate it here would be the usage of crossfilter groups objects that we pass to the dc.js which would not look eloquent as well. Yes, I'll try to think about a better solution.

You're right about the breakout series. Then there are two X-axis rows selected and both have unaggregated values it does not aggregate Y value by both. I'll create another issue since it is not a regression and a rarer case.

Thanks for your review!

**Pull request merged**
